File: backend/src/main/java/com/back/domain/book/book/dto/BookSearchDto.java
package com.back.domain.book.book.dto;

import com.back.domain.bookmarks.constant.ReadState;
import lombok.Builder;
import lombok.Getter;

import java.time.LocalDateTime;
import java.util.List;

@Getter
@Builder
public class BookSearchDto {
    private int id;
    private String title;
    private String imageUrl;
    private String publisher;
    private String isbn13;
    private int totalPage;
    private LocalDateTime publishedDate;
    private float avgRate;
    private String categoryName;
    private List<String> authors;
    private ReadState readState;
}

File: backend/src/main/java/com/back/domain/book/book/dto/BookSearchDto.kt
package com.back.domain.book.book.dto

import com.back.domain.bookmarks.constant.ReadState
import java.time.LocalDateTime

data class BookSearchDto(
    val id: Long = 0,
    val title: String? = null,
    val imageUrl: String? = null,
    val publisher: String? = null,
    val isbn13: String? = null,
    val totalPage: Int = 0,
    val publishedDate: LocalDateTime? = null,
    val avgRate: Float = 0f,
    val categoryName: String? = null,
    val authors: List<String> = emptyList(),
    val readState: ReadState? = null
)